src/core/qgsfield.cpp

```diff
@@ -258,6 +281,13 @@ bool QgsField::convertCompatible( QVariant &v ) const
     return true;
   }
 
+  // Give it a chance to convert to double since we accept both comma and dot as decimal point
```

Comment:
I think this should only be applied for locales which use comma as decimal... it may have odd implications for dot as decimal locales with thousand separated strings